Issue #13345: Enabled OuterTypeFilenameCheckExamplesTest

MANISH-K-07 opened the pull request:
Part of #13345 

Enabled OuterTypeFilenameCheckExamplesTest

sktpy commented:
@MANISH-K-07  Please link the documentation for the future PRs. ([Documentation](https://checkstyle.sourceforge.io/checks/misc/outertypefilename.html))

sktpy commented:
GitHub, generate website

sktpy commented on src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/outertypefilename/Example3.java:
```diff
@@ -5,9 +5,14 @@
 
 // xdoc section -- start
-class Foo { // violation
+// public class Example1 {
+// // This class is commented out
+// }
```
```suggestion
```
Please remove all such occurrences or explain why it is need.

sktpy commented on src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/outertypefilename/Example5.java:
```diff
@@ -0,0 +1,18 @@
+// // This class is commented out
+// }
+
+class Foo5 { // violation, 'The name of the outer type and the file do not match.'
+
+}
```
Documentation uses `record` for this example. Is there any reason why you have decided to go with class instead?
You probably were facing issues because you weren't placing it in `resources-noncompilable`.

Refer to the PR #13275  for reference. (Particularly look for how `Example1.java` is used from `resources-noncompilable`, update the test accordingly.)

sktpy commented on src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/OuterTypeFilenameCheckExamplesTest.java:
```diff
@@ -37,42 +37,42 @@ public void testExample1() throws Exception {
 
         };
```
You should use `CommonUtil.EMPTY_STRING_ARRAY` for empty string arrays for the purpose of clarity and consistency with the rest of codebase.

MANISH-K-07 commented:
@sktpy ,
Thank you for showing interest on my PR.
I'm not sure if you have noticed but there are multiple pushes to this PR. That is because I have already tried everything you suggested times two plus more !!

**_I have been working with checkstyle and understanding its codebase and CIs since an ample amount of time...._**
Please try implementing your said fixes before you suggest them out of the blue.

> Please remove all such occurrence or explain why it is needed.

A class with name matching the pattern `"Example + d"` is must even in a non-compilable path.

> Documentation uses record for this example. Is there any reason why you have decided to go with class instead?
> You probably were facing issues because you weren't placing it in resources-noncompilable.

It is true that there were issues with the `record` keyword since it is only compilable on java 17. But moving it to a non-compilable path doesn't work either.
The Checkstyle checks do not encourage the usage of the word itself !!
Checkstyle throws an Exception which causes the CIs to fail.

Th only way to satisfy the above two is to add a suppression which I don't think is preferrable until suggested by maintainers.
> You should use CommonUtil.EMPTY_STRING_ARRAY for empty string arrays for the purpose of clarity and consistency with the rest of codebase.

I don't know if you have observed the rest of the code but almost all the example checks do not use this import.
_The Reason being :_ 
The example checks are way too small and numerous. Importing to every single file just for the purpose of a single test method or two causes a spike in ample amount of millis which I don't think is preferrable in any end-user !!

That being said, if the maintainers have a different perspective on this, I will definitely make changes accordingly as they are much more experienced than we ever could be !!

sktpy commented:
@MANISH-K-07 Hi! I haven't recommended these changes before trying them out myself.
I have done each and every one of them and they work without any issue. **If you want**, I can guide you more to where you are getting it wrong.

I know that you have been contributing here before me, but we can all learn from each other.
But if you rather only want suggestions and advice from maintainers only, I would remember to not bother you anytime in the future.
Thankyou for your time. :heart:

MANISH-K-07 commented:
@sktpy ,
Hey !! That is totally not what I meant. 
I would love more than anything to learn new stuff and if your suggestion is providing me with that, I am all ears. 

The thing is I initially began this PR with the exact same thought of what you suggested and error-by-error, I evolved it into what it is at present. So, seeing the same changes, I was not sure if any of it would work (my mistake).

It would be great if you could have a look at these commits (outdated) and then suggest in case there's anything wrong : 

- 5a9294d 
-   a3a0e90 
-  a5a7e21 

I had the files moved to non-compilable path and kept the `record` keyword intact. There was checkstyle exception and the CIs obviously failed. Replacing the record with a normal class gave me the errors related to example name matching. So, I added the commented classes to satisfy the CIs again. After doing all this, there was no reason for the files to be non-compilable, so I moved them back to compilable finally resulting in the PR that you see.

Apart from all this.... your suggestions are always open. I didn't mean it that way and Sorry if it sounded like that
**`:)`**

sktpy commented:
@MANISH-K-07 Thank you for understanding. 
1. You do not need all the files to be in non-compileable directory. In this case you only need to have one with `record` in that, see the PR that I have referenced in the corresponding review.
2. Get familiar with the record class to make sure you are not making any syntax error. ([Reference](https://web.archive.org/web/20240119085414/https://docs.oracle.com/en/java/javase/17/language/records.html#GUID-6699E26F-4A9B-4393-A08B-1E47D4B2D263))
3. Regarding the one for removal of comments, try to be more creative with it, and incorporate a class matching the format in code that feels natural to exist there. (like if you see the website generated by github above, the commented class all over the place does not appear right) 
[unless maintainers agree for some changes in the check that is producing that error, or exclude certain examples]

Feel free to reach out, and ask for further help. I can provide exact code but I would recommend that as a last resort, because that would defeat the purpose of these good first PRs.

MANISH-K-07 commented:
> You do not need all the input files to be in non-compileable directory. In this case you only need to have one with record in that, see the PR that I have referenced in the corresponding review.

Yepp, tried that too. didn't push bcoz it was still failing.

> Get familiar with the record class to make sure you are not making any syntax error.

Sure I will do that. However, the error wasn't syntactical or compilation error. It was a CheckstyleException. Checkstyle isn't simply allowing the use of the keyword record in non-compilable path either. I am curious as to how did you manage a workaround to that ?....

> Regarding the one for removal of comments, try to be more creative with it, and incorporate a class matching the format in code that feels natural to exist there.

That really sounds good to hear !!
though I can't think of any possible adjustment as there is only one single line of code in the file. If you have any idea to modify such that both beautification and CI satisfaction could be covered, then please do share.

sktpy commented:
> Yepp, tried that too. didn't push bcoz it was still failing.

Can you link the commit (push it on separate branch perhaps) with the error, so that I can pin point the issue.
<br>

> I can't think of any possible adjustment as there is only one single line of code in the file

You don't have to limit yourself to the single line stated in example, if that means you can make it more presentable (and get rid of that commented class in each of the example). Any creative change probably will be more presentable.
https://github.com/checkstyle/checkstyle/pull/14363#pullrequestreview-1848976960 "formatting should look good"
Although, I would recommend it to keep it within reasonable limits so that examples are not too much bloated.

MANISH-K-07 commented:
GitHub, generate website

romani commented on src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/outertypefilename/Example2.java:
```diff
@@ -0,0 +1,18 @@
+// xdoc section -- start
+// public class Example1 {
+// // This class is commented out
+// }
```
please remove such commented out code.
we have description over example like `Example of class Foo2 in a file named Example2.java` so it is enough to understand an problem for user.

MANISH-K-07 replied:
I added the commented class to satisfy the `exampleClassName` check.
if we remove those, we will have to add suppressions.

Should I proceed with adding new suppressions ?

romani replied:
Yes suppression, as this module by design needs different name.
Or you can try make other class to be name Example1ButNotSameName it might pass validation:)

MANISH-K-07 replied:
> Or you can try make other class to be name `Example1ButNotSameName `it might pass validation:)

Cheat-code applied successfully  ;)

romani reviewed (changes requested):
items:

MANISH-K-07 commented:
GitHub, generate website

romani commented on src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/outertypefilename/Example2.java:
```diff
@@ -0,0 +1,15 @@
+
+}
+class Example2ButNotSameName {}
+// xdoc section -- end
```
```
// xdoc section -- start
class Example2ButNotSameNameAsFile {}  // violation, 'The name of the outer type and the file do not match.'
// xdoc section -- end
```

MANISH-K-07 replied:
Done....
also removed the empty block space for all other example files like you suggested in my other PR, hope its all good.

romani replied:
Please update all other examples to use this pattern.
And please update text above example , see generated website page, there should be no Foo anymore

MANISH-K-07 replied:
Made the changes..
Please review if it looks good now

romani reviewed (changes requested):
items

romani commented:
GitHub, generate website

romani commented:
See comment above

MANISH-K-07 commented:
GitHub, generate website

romani commented on src/xdocs/checks/misc/outertypefilename.xml:
```diff
@@ -27,35 +27,35 @@
         </source>
-        <p id="Example1-code">Example of class Test in a file named Test.java</p>
+        <p id="Example1-code">
+          Example of class Example1 in a file named Example1.java
```
please change wording to avoid class name, we should use "file content".
`Example file content with name of file Example1.java`

please apply to all others.

MANISH-K-07 replied:
Done 😄

romani reviewed (changes requested):
last:

MANISH-K-07 commented:
GitHub, generate website

romani reviewed (approved):
Ok to merge
